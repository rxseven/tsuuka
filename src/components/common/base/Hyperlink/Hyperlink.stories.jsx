import React from 'react';
import { storiesOf } from '@storybook/react';

import Hyperlink from './index';

const title = 'Components/Common/Base/Hyperlink';
const props = { children: 'Link Text', href: '/' };
const component = <Hyperlink {...props} />;

storiesOf(title, module).add('default', () => component);
