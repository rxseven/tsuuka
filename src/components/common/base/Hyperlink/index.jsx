import PropTypes from 'prop-types';
import React from 'react';

const propTypes = {
  children: PropTypes.node.isRequired,
  href: PropTypes.string.isRequired,
  rel: PropTypes.string,
  replace: PropTypes.bool
};

const defaultProps = {
  rel: 'noopener noreferrer',
  replace: false
};

function Hyperlink({ children, replace, ...props }) {
  return (
    <a {...props} target={replace ? '_self' : '_blank'}>
      {children}
    </a>
  );
}

Hyperlink.propTypes = propTypes;
Hyperlink.defaultProps = defaultProps;

export default Hyperlink;
