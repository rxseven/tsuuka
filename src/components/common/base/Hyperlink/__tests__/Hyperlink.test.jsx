import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import Hyperlink from '../index';

describe('<Hyperlink />', () => {
  // Arrange
  const props = { children: 'Link Text', href: '/' };
  const component = <Hyperlink {...props} />;

  describe('Unit tests', () => {
    it('should render without crashing', () => {
      // Act
      const wrapper = shallow(component);

      // Assert
      expect(wrapper).toBeDefined();
    });
  });

  describe('Snapshot tests', () => {
    it('should render correctly', () => {
      // Act
      const tree = renderer.create(component).toJSON();

      // Assert
      expect(tree).toMatchSnapshot();
    });
  });
});
