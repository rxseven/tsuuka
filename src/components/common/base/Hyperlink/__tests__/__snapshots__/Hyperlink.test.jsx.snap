// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hyperlink /> Snapshot tests should render correctly 1`] = `
<a
  href="/"
  rel="noopener noreferrer"
  target="_blank"
>
  Link Text
</a>
`;
